web/Controllers/AbstractController.cs

```diff
@@ -30,4 +30,18 @@ protected void FlashMessage(ILogger logger, (string logInfo, string flashtype, s
         TempData["FlashType"] = flash.flashtype;
         TempData["FlashMessage"] = flash.flashmessage;
     }
+    
+    protected void FlashMultipleMessages(ILogger logger, (List<string> logInfo, List<string> flashtypes, List<string> flashmessages) flashes)

```

Comment:
I think take the logger out.

We should log things as they happen to help with debugging, whereas this will run at the end.